### PR TITLE
Makes AcceptedConnectionsLimit configurable

### PR DESCRIPTION
**Pull request by bolt12**

### CODE REVIEW

File: cardano-node/src/Cardano/Node/Configuration/POM.hs [hunk 1]
```diff
@@ -420,6 +432,14 @@ defaultPartialNodeConfiguration =
...
     , pncProtocolIdleTimeout   = Last (Just 5)
     , pncTimeWaitTimeout       = Last (Just 60)
+    , pncAcceptedConnectionsLimit =
+      Last
+      $ Just
+      $ AcceptedConnectionsLimit
+        { acceptedConnectionsHardLimit = 512
+        , acceptedConnectionsSoftLimit = 384
+        , acceptedConnectionsDelay     = 5
+        }
```
Comment (coot) [inline cardano-node/src/Cardano/Node/Configuration/POM.hs:442]:
```suggestion
        Last
      $ Just
      $ AcceptedConnectionsLimit
        { acceptedConnectionsHardLimit = 512
        , acceptedConnectionsSoftLimit = 384
        , acceptedConnectionsDelay     = 5
        }
```

File: cardano-node/src/Cardano/Node/Orphans.hs [hunk 2]
```diff
@@ -42,3 +44,26 @@ instance FromJSON Update.ApplicationName where
...
+          , acceptedConnectionsDelay
+          } =
+    object [ "AcceptedConnectionsLimit" .=
+      object [ "acceptedConnectionsHardLimit" .=
+                  toJSON acceptedConnectionsHardLimit
+             , "acceptedConnectionsSoftLimit" .=
+                  toJSON acceptedConnectionsSoftLimit
+             , "acceptedConnectionsDelay" .=
+                  toJSON acceptedConnectionsDelay
+             ]
```
Comment (coot) [inline cardano-node/src/Cardano/Node/Orphans.hs:61]:
```suggestion
      object [ "hardLimit" .=
                  toJSON acceptedConnectionsHardLimit
             , "softLimit" .=
                  toJSON acceptedConnectionsSoftLimit
             , "delay" .=
                  toJSON acceptedConnectionsDelay
             ]
```

File: cardano-node/src/Cardano/Node/Orphans.hs [hunk 3]
```diff
@@ -42,3 +44,26 @@ instance FromJSON Update.ApplicationName where
...
+           ]
+
+instance FromJSON AcceptedConnectionsLimit where
+  parseJSON = withObject "AcceptedConnectionsLimit" $ \v ->
+    AcceptedConnectionsLimit
+      <$> v .: "acceptedConnectionsHardLimit"
+      <*> v .: "acceptedConnectionsSoftLimit"
+      <*> v .: "acceptedConnectionsDelay"
```
Comment (coot) [inline cardano-node/src/Cardano/Node/Orphans.hs:69]:
```suggestion
      <$> v .: "hardLimit"
      <*> v .: "softLimit"
      <*> v .: "delay"
```

**Review by coot — APPROVED** [review]
LGTM, just some minor suggestion.

**Comment by bolt12** [comment]
bors merge

**Review by Jimbo4350 — APPROVED** [review]

**Comment by bolt12** [comment]
bors merge

**Comment by bolt12** [comment]
closes https://github.com/input-output-hk/cardano-node/pull/3435

**Pull request merged**